src/sentry/testutils/factories.py

```diff
@@ -262,11 +266,31 @@ def _patch_artifact_manifest(path, org=None, release=None, project=None, extra_f
 class Factories:
     @staticmethod
     @assume_test_silo_mode(SiloMode.REGION)
-    def create_organization(name=None, owner=None, **kwargs):
+    def create_organization(name=None, owner=None, region: Region | str | None = None, **kwargs):
         if not name:
             name = petname.generate(2, " ", letters=10).title()
 
-        org = Organization.objects.create(name=name, **kwargs)
+        if isinstance(region, str):
+            region = get_region_by_name(region)
+
+        @contextmanager
+        def org_creation_context():
+            if region is None:
+                yield
+            else:
+                with override_settings(SILO_MODE=SiloMode.REGION, SENTRY_REGION=region.name):
+                    with override_regions([region]):
```

Comment:
I want to call out this part in particular. It might look more verbose than it needs to be, i.e., above and beyond what we typically handle with `assume_test_silo_mode(SiloMode.REGION)`. But AFAICT it's necessary because the snowflake ID generation is going to look for the local region.

There are preexisting test cases where we were overriding the set of existing regions, but not `SENTRY_MONOLITH_REGION` because it wasn't relevant to the test case. This caused a bogus `--monolith--` region to float around in the global directory. When it came time to assign a snowflake ID, the code would resolve to the `--monolith--` region by default. Now we're cleaning up our act and eliminating the `--monolith--` region from contexts where it shouldn't exist, but -- oops! -- that means the snowflake code needs more context in order to work.

I believe these are good improvements, in that they make the test environment behave a little bit more like prod. But I'm open to any suggestions for a less heavy-handed way to set it up.

Reply:
With regard to this being excessively heavy-handed, `with override_regions([region])` in particular is a symptom of not obeying the invariant "the region map should be the same at all points during each run" as described in [my comment below](https://github.com/getsentry/sentry/pull/52768#issuecomment-1636289548).

Reply:
👍  I'm ok with this approach.  Minor nit but you can fit multiple contexts on the same indention, ie: `with override_settings(), override_regions():`

Reply:
The `with override_regions([region])` kludge is now moot, thanks to changes in https://github.com/getsentry/sentry/pull/53028 😌

> 👍 I'm ok with this approach. Minor nit but you can fit multiple contexts on the same indention, ie: `with override_settings(), override_regions():`

I know. 😅 That was a stylistic choice because the Black autoformatting insisted on putting line breaks in terrible places.